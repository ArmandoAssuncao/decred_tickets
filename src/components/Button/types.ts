export interface IProps {
    m: string
    onPress: () => void
    title: string
}

export interface IState {
}
