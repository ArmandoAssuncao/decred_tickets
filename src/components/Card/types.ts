import { ITicket } from 'store/ticket/types'

export interface IProps {
    onPress?: () => void
    type: ITicket['status']
    value: number
}

export interface IState {
}
