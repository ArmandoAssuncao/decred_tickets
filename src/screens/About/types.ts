import { IScreenProps } from 'containers/Root/types'

export interface IConnectedProps {
}

export interface IActionCreators {
}

export interface IProps extends IScreenProps, IConnectedProps, IActionCreators {
}

export interface IState {
}
