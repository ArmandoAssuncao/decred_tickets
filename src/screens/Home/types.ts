import { IScreenProps } from 'containers/Root/types'
import { getTicket } from 'store/ticket/actions'
import { ITicketState } from 'store/ticket/types'

export interface IConnectedProps {
    ticket?: ITicketState
}

export interface IActionCreators {
    getTicket: typeof getTicket
}

export interface IProps extends IScreenProps, IConnectedProps, IActionCreators {
}

export interface IState {
    stakePool: ITicketState['stakePool']
    ticketAddress: ITicketState['address']
}
