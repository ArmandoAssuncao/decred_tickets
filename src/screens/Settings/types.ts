import { IScreenProps } from 'containers/Root/types'
import { permitNotifications } from 'store/settings/actions'
import { ISettingsState } from 'store/settings/types'

export interface IConnectedProps {
    settings?: ISettingsState
}

export interface IActionCreators {
    permitNotifications: typeof permitNotifications
}

export interface IProps extends IScreenProps, IConnectedProps, IActionCreators {
}

export interface IState {
    isNotifications: boolean
}
