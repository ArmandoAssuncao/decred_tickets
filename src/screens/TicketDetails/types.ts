import { IScreenProps } from 'containers/Root/types'
import { ITicket } from 'store/ticket/types'

export interface IConnectedProps {
}

export interface IActionCreators {
}

export interface IProps extends IScreenProps<INavigationState>, IConnectedProps, IActionCreators {
}

export interface IState {
}

export interface INavigationState {
    params: {
        tickets: ITicket[]
    }
}
