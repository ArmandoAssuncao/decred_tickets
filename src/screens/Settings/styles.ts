import { TextStyle, ViewStyle } from 'react-native'
import EStyleSheet from 'react-native-extended-stylesheet'

export default EStyleSheet.create({
    container: {
        backgroundColor: '$primaryColor',
        flex: 1,
    } as ViewStyle,
    scrollContentContainer: {
        padding: '0.4rem',
    } as ViewStyle,
    text: {
        color: '$textColorSecond',
    } as TextStyle,
    notificationContainer: {
        flexDirection: 'row',
        justifyContent: 'space-around',
    } as ViewStyle,
})
