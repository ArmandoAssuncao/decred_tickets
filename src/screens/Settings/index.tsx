import React from 'react'
import { ScrollView, Switch, View } from 'react-native'
import { NavigationScreenConfig } from 'react-navigation'
import { connect, MapDispatchToProps } from 'react-redux'
import { bindActionCreators } from 'redux'
import OneSignal from 'react-native-onesignal'
import { IAppState } from 'store/types'

import i18n from 'config/i18n'
import { permitNotifications } from 'store/settings/actions'
import { Text } from 'components'

import { IActionCreators, IConnectedProps, IProps, IState, } from './types'
import styles from './styles'

const mapStateToProps = (state: IAppState): IConnectedProps =>
  ({
      settings: state.settings,
  })

const mapDispatchToProps = (dispatch: any): MapDispatchToProps<IActionCreators, IProps> =>
    bindActionCreators(
        {
            permitNotifications,
        },
        dispatch
    )

export class Settings extends React.PureComponent<IProps, IState> {
    static navigationOptions: NavigationScreenConfig<any> = {
        title: i18n.t('screens.settings.title'),
    }

    constructor(props: IProps) {
        super(props)

        this.state = {
            isNotifications: props.settings.permitNotifications,
        }
    }

    private onChangeNotifications = (value: boolean): void => {
        this.setState({ isNotifications: value })
        this.props.permitNotifications(value)
        OneSignal.setSubscription(value)
    }

    render(): JSX.Element {
        const { isNotifications } = this.state
        return (
            <ScrollView style={styles.container} contentContainerStyle={styles.scrollContentContainer}>
                <View style={styles.notificationContainer}>
                    <Text
                        id={'screens.settings.notifications'}
                        m="Notifications"
                        type="bold"
                        style={styles.text}
                    />
                    <Switch value={isNotifications} onValueChange={this.onChangeNotifications} />
                </View>
            </ScrollView>
        )
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(Settings)
