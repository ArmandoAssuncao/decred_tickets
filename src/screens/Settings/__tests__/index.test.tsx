import * as React from 'react'
import { shallowStore } from 'test/enzyme'

import Settings from '../index'
import { ISettingsState } from 'store/settings/types'

describe('Settings test render', () => {
    const store = {
        settings: {
            permitNotifications: true,
        } as ISettingsState,
    }

    const wrapper = shallowStore(<Settings permitNotifications={jest.fn()} />, store)

    it('Testing snapshot', () => {
        expect(wrapper).toMatchSnapshot()
    })

    it('Testing instance of component', () => {
        expect(wrapper.instance()).toBeInstanceOf(Settings.WrappedComponent)
    })

    it('has 1 ScrollView', () => {
        expect(wrapper.find('ScrollView')).toHaveLength(1)
    })
})
