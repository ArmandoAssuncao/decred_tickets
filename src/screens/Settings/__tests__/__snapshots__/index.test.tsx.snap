// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings test render Testing snapshot 1`] = `
<ScrollView
  contentContainerStyle={
    Object {
      "padding": 6.4,
    }
  }
  style={
    Object {
      "backgroundColor": "#0C1E3E",
      "flex": 1,
    }
  }
>
  <Component
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-around",
      }
    }
  >
    <Text
      id="screens.settings.notifications"
      m="Notifications"
      size="md"
      style={
        Object {
          "color": "#FFF",
        }
      }
      type="bold"
    />
    <Switch
      disabled={false}
      onValueChange={[Function]}
      value={true}
    />
  </Component>
</ScrollView>
`;
